docs/configuration.md

```diff
@@ -0,0 +1,350 @@
+a look at the [LabelSelector docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements)
+
+A common selector for worker nodes looks like this:
+```yaml
+selector:
+  matchExpressions:
+    - key: node-role.kubernetes.io/control-plane
+      operator: DoesNotExist
+    - key: node-role.kubernetes.io/master
+      operator: DoesNotExist
+```
+The reason for excluding `master` and `control-plane` nodes instead of selecting
+`worker` nodes is, that this also prevents potentially unwanted remediation of
+control plane nodes in compact clusters, where nodes have both roles.
+
+For remediating control plane nodes use this:
+```yaml
+selector:
+  matchExpressions:
+    - key: node-role.kubernetes.io/control-plane
+      operator: Exists
+```
+> **Note**
+>
+> On older clusters you have to use `master` in above example. You can't use
+> both `master` and `control-plane` because the expressions are evaluated with
+> logical "AND".
+
+> **Warning**
+>
+> - Having a configuration which selects both worker and control plane nodes
+    > is strongly discouraged, because control plane nodes have special handling
+    > in NHC and potentially in remediators!
+> - Multiple configurations must not select an overlapping node set! This can lead to unwanted remediations.
+
+### RemediationTemplate
+
+The remediation template is an [ObjectReference](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-reference/)
+to a remediation template provided by a remediation provider. Mandatory fields
+are `apiVersion`, `kind`, `name` and `namespace`.
+
+> **Note**
+> 
+> This field is mutually exclusive with spec.EscalatingRemediations
+
+Note that some remediators work with the template being created in any namespace,
+others require it to be in their installation namespace.
+
+Also, some remediators install a remediation template by default, which can be
+used by NHC. This is e.g. the case for the SelfNodeRemediation remediator. The
+example CR above shows how to use its "resource deletion strategy" template.
+
+For other remediators you might need to create a template manually. Please check
+their documentation for details.
+
+For more details on the remediation template, and the remediation CRs created
+by NHC based on the template, see [below](#remediation-resources)
+
+### EscalatingRemediations
+
+EscalatingRemediations is a list of RemediationTemplates with an order and
+timeout field. Instead of just creating one remediation CR and waiting forever
+that the node gets healthy, using this field offers the ability to to try
+multiple remediators one after another.
+The `order` field determines the order in which the remediations are invoked
+(lower order = earlier invocation). The `timeout` field determines when the
+next remediation is invoked.
+
+There are optional features available when using escalating remediations:
```

Comment:
NIT: I can't recall if is recommended to begin a sentence with a capital letter or not after a bullet, but we are incosistent on this page.